dsgrid/config/time_dimension_base_config.py

```diff
@@ -210,19 +211,20 @@ def _convert_time_to_project_time(
         - wrap_time_allowed from InputDatasetModel is used to align time due to
         time_zone differences
         """
-        # Persist the table because the code below will need to evaluate the query multiple times.
-        # TODO: Do we still need this after Lixi's improvements?
-        df = persist_intermediate_table(df, context)
-
         if time_based_data_adjustment is None:
             time_based_data_adjustment = TimeBasedDataAdjustmentModel()
 
+        # It's possible that we will want to enable this in some cases.
+        # Current results show about the same timings with and without.
+        # df = persist_intermediate_table(df, context)
+
         time_col = project_time_dim.get_load_data_time_columns()
         assert len(time_col) == 1, time_col
         time_col = time_col[0]
         time_map = (
             df.select(time_col).distinct().select(time_col, F.col(time_col).alias("orig_ts"))
-        )
+        ).cache()
+        time_map.count()
```

Comment:
What does this do?

Reply:
Calling df.cache() by itself does nothing. If you call df.count(), it will materialize the current table.

Reply:
Maybe we materialize just before time_map_diff?

Reply:
`df` is a very large table. The query represented by `time_map` reduces that table to 8760/8784 rows with one column. We should materialize it as soon as possible. Otherwise, that query will be evaluated in each of `_align_time_interval_type`, `filter_to_project_timestamps`, `time_difference`, and `apply_time_wrap`. Or, am I missing your point?